### DISCUSSION TITLE
Checkbox + text component

### DISCUSSION THREAD

**Discussion by JoseGonzalz** [Telefonica/mistica-design#1644] [Proposals]
### Goal

I would like to propose the creation of several options for this component.

### Context

Currently, this component seems to have been created for legal text, specifically to accept terms and conditions, that is why it is aligned to the left.
<img width="148" alt="Captura de pantalla 2024-02-09 a las 12 30 48" src="https://github.com/Telefonica/mistica-design/assets/129079621/f1ebbf79-b0c5-4d65-bda2-bed1326d4ef2">

The Brazilian team have created new context for this. They are using it to select/deselect all bills and the current component is unlikely to fit the need.
![Frame 1171277699](https://github.com/Telefonica/mistica-design/assets/129079621/fa559c5a-3bac-454a-b0b4-224d7fd5581f)

The DIPA team need to globalise the experience implemented in VIVO in each OB (MOVES, O2ES) however if I use the Mística component, I get this
![Debt payment-Moves-Debt selector-02_00-mixed](https://github.com/Telefonica/mistica-design/assets/129079621/284bb347-8310-4fbc-87a1-b4a5eda49b2e)

### Impact

Some reasons:
- Give the Mística component to VIVO because they are using an ad hoc component.
- Be able to be used in other OBs, such as MOVES and O2ES, to globalise the same experience.

### Discussion Questions

We should have different options to adapt to the context.
- Checkbox + txt
- Txt + checkbox

And to be able to have different text style.
- For terms and conditions
- For secondary actions (like this one)

![e3](https://github.com/Telefonica/mistica-design/assets/129079621/c4f140d3-7a66-4a10-af3c-292d35c8eae0)

**Comment by yceballost** [comment]
Imo, makes sense to evolve checkboxes to allow label on the left and right side

The component right now allows changes in text style (colors and sizes) (but I believe that sizing should only be done in very justified situations) 
[Example →](https://rebrand.ly/tina7vm)

Also I propose to change the text of the component in figma to "[ ] Label" or "[ ] Label link"

**Comment by AnaMontes11** [comment]
Hello @JoseTelefonica, after discussing it with the development team, as of today, that possibility is already configurable. Therefore, for now, you can build it atomically (Text + checkbox). We'll make a note on our end to include it in the future, but for this quarter, we have to leave it out of the backlog.
[Example](https://mistica-web.vercel.app/playroom/#?code=N4Igxg9gJgpiBcIA8AhCAPABABwIZSgEsA7AcwF5gBmAJgF8A%2BAHWM0yQGUAXXMAa0wBnPGBiUAjADZGLNkgDCACxj8ARhgYAVGOi6YAToVKKuSAPRKVfdemat2ASWIAbEjCEixwKYyTbdVFo6es4wAGamZv5cgQrKahhmDOZOrsQwdmxylgm20QZGJuzRADIkfAwAtgCemGl85qXlDOwW8dYaso4ubh68Xj7J0YFdWVn5oRHFwWXEFTV15Y0zzaNZywHJOR3oSSk96Xbm3LwVLOZotiAANCBcypUwgggA2iAAshAAboSCPPogAC6twA7oQoPdnvAXuIAAw0AAsgLoQA)

**Reply by AnaMontes11** [reply]
Hello again @JoseTelefonica 
Although it can be done atomically today, we will enable it to come by default in the component. 
I close the discussion and link it to issue #1658.
